shell.nix

```diff
@@ -0,0 +1,23 @@
+let # Rust
+    rust_overlay = import (builtins.fetchTarball "https://github.com/oxalica/rust-overlay/archive/master.tar.gz");
+    pkgs = import <nixpkgs> { overlays = [ rust_overlay ]; };
+    rustVersion = "1.75.0";
+    rust = pkgs.rust-bin.stable.${rustVersion}.default.override {
+      extensions = [
+        "rust-src" # for rust-analyzer
+      ];
+    };
```

Comment:
If you want to use newer rustc, you could instead import nixpkgs-unstable from some date (like today). Not sure which way is better. I usually use the method with unstable.

Reply:
Example:
```
import (builtins.fetchTarball {
  # Descriptive name to make the store path easier to identify
  name = "nixos-unstable-2018-09-12";
  # Commit hash for nixos-unstable as of 2018-09-12
  url = "https://github.com/nixos/nixpkgs/archive/ca2ba44cab47767c8127d1c8633e2b581644eb8f.tar.gz";
  # Hash obtained using `nix-prefetch-url --unpack <url>`
  sha256 = "1jg7g6cfpw8qvma0y19kwyp549k1qyf11a5sg6hvn6awvmkny47v";
}) {}
```

Reply:
Me neither, so let's leave as is